### PR TITLE
CB-14886. Run metering checks as part of preflight checks during depl…

### PR DESCRIPTION
**Pull request by oleewere**
…oyment/diagnostics

details:
- update usage.proto and regenerate java classes
- extends diagnostics flow service to run metering checks
- outputs are gathered and logged, also a cdp diagnostics usage is sent based on the results

See detailed description in the commit message.

### CODE REVIEW

**Comment by oleewere** [comment]
retest

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java [hunk 1]
```diff
@@ -167,6 +173,82 @@ public void reportNetworkCheckUsages(Long stackId, List<NodeStatusProto.NetworkD
...
+                    String resourceCrn = stack.getResourceCrn();
+                    String accountId = Crn.safeFromString(resourceCrn).getAccountId();
+                    UsageProto.CDPDiagnosticEvent.Builder diagnosticsEventBuilder = UsageProto.CDPDiagnosticEvent.newBuilder();
+                    diagnosticsEventBuilder
+                            .setAccountId(accountId)
+                            .setResourceCrn(resourceCrn)
+                            .setEnvironmentCrn(stack.getEnvironmentCrn())
+                            .setServiceType(UsageProto.ServiceType.Value.DATAHUB);
+                    if (errorSet.isEmpty()) {
+                        diagnosticsEventBuilder.setResult(UsageProto.CDPDiagnosticResult.Value.UNSET);
```
Comment (vaibhavgandhi12) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java:219]: Don't send this if there is no issue.

**Review by vaibhavgandhi12 — COMMENTED** [review]
Looks like a great start. I would love to see this run periodically, eventually.

**Review by lnardai — APPROVED** [review]

**Review by vaibhavgandhi12 — APPROVED** [review]
lgtm!

**Review by nadamdb — APPROVED** [review]
LGTM

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java [hunk 2]
```diff
@@ -167,6 +173,82 @@ public void reportNetworkCheckUsages(Long stackId, List<NodeStatusProto.NetworkD
...
+        try {
+            Stack stack = stackService.getById(stackId);
+            if (stack.isDatalake()) {
+                LOGGER.debug("Skip metering check as billing is not used for datalake");
+                return;
+            }
+            if (!meteringConfiguration.isEnabled()) {
+                LOGGER.debug("Metering feature is not enabled (or telemetry is empty). Skip metering check.");
+                return;
+            }
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java:186]: please don't do empty returns, use else and refactor your code please

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java [hunk 3]
```diff
@@ -167,6 +173,82 @@ public void reportNetworkCheckUsages(Long stackId, List<NodeStatusProto.NetworkD
...
+                    reportMeteringUsage(nodeStatuses, NodeStatusProto.MeteringDetails::getHeartbeatConfig,
+                            "Heartbeat agents configured", errorSet);
+                    reportMeteringUsage(nodeStatuses, NodeStatusProto.MeteringDetails::getLoggingServiceRunning,
+                            "Logging agents running", errorSet);
+                    reportMeteringUsage(nodeStatuses, NodeStatusProto.MeteringDetails::getLoggingAgentConfig,
+                            "Logging agents configured", errorSet);
+                    reportMeteringUsage(nodeStatuses, NodeStatusProto.MeteringDetails::getDatabusReachable,
+                            "DataBus reachable", errorSet);
+                    reportMeteringUsage(nodeStatuses, NodeStatusProto.MeteringDetails::getDatabusTestResponse,
+                            "DataBus test response", errorSet);
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/DiagnosticsFlowService.java:205]: this seems a bit C style, would be more object oriented if you would have like an interface/abstract class with multiple implementation all of which would be called and could be injected by spring framework

**Review by lacikaaa — CHANGES_REQUESTED** [review]

**Comment by lacikaaa** [comment]
as this is merged please fix these in a followup

**Pull request merged**